src/mono/mono/mini/interp/interp.c

```diff
@@ -3420,7 +3457,9 @@ interp_exec_method (InterpFrame *frame, ThreadContext *context, FrameClauseArgs
 
 			csignature = (MonoMethodSignature*)frame->imethod->data_items [ip [4]];
 
-			cmethod = LOCAL_VAR (ip [2], InterpMethod*);
+			/* In mixed mode, stay in the interpreter for simplicity even if there is an AOT version of the callee */
+			cmethod = ftnptr_to_imethod (LOCAL_VAR (ip [2], gpointer));
```

Comment:
This makes CALLI extremely slow when using the function pointer only in the interpreter since `mono_interp_get_imethod` takes lock. Maybe we should cache the imethod in the descriptor ?